projects/eval-pqmon/src/common/iio_pqm.c

```diff
@@ -514,11 +551,220 @@ int read_ch_attr(void *device, char *buf, uint32_t len,
 				       buf, len, "%" PRIu32 "",
 				       desc->pqm_ch_attr[channel->ch_num + VOLTAGE_CH_NUMBER][attr_id]);
 		}
-
+	case IIO_COUNT:
+		switch (channel->ch_num) {
+		case 0: // Dips
+			switch (attr_id) {
+			case CHAN_EVENT_COUNT:
+				return snprintf(buf, len, "%" PRIu8 "", pqlibExample.output->events.dipCount);
+			case CHAN_EVENT_START_TIME:
+				strcpy(buf, "");
```

Comment:
There are lots of these code blocks (repeated 16 times if i counted correctly)
https://github.com/analogdevicesinc/no-OS/pull/2360/commits/3301042309a606d7e60be2ee07fac0198bb958df#diff-65295bfa3bf785bc58b4f379a8a4425e0c7841967111db9bc87e26629a317e42R561-R569

Could you do a function that holds that for loop ? I think you'd need buf, pqLibExample and attr_id as arguments for the function. This would simplify this giant switch case a lot.